source/gloperate/include/gloperate/pipeline/Stage.h

```diff
@@ -652,6 +652,59 @@ class GLOPERATE_API Stage : public cppexpose::Object
     template <typename T>
     void forAllOutputs(std::function<void(gloperate::Output<T> *)> callback);
 
+	/**
+	*  @brief
+	*    The time spent in onProcess in nanoseconds.
+	*
+	*  @return
+	*    unsigned long representing cpu duration of onProcess in nanoseconds
+	*/
+	std::uint64_t lastCPUTime() const;
+
+	/**
+	*  @brief
+	*    The gpu time for gpu commands issued during onProcess in nanoseconds.
+	*
+	*  @return
+	*    unsigned long representing gpu duration of onProcess in nanoseconds
+	*
+	*  @remarks
+	*    Due to the async nature of gpu processing, this value is one iteration (i.e. frame) late.
+	*/
+	std::uint64_t lastGPUTime() const;
+
+	/**
+	*  @brief
+	*    Return the history of cpu time measurement of onProcess (most recent values last).
+	*
+	*  @remarks
+	*    Complexity is linear in the size of the history.
```

Comment:
Does this complexity count for the accessor or the internal management during measurements?